File: LQD/MAKESTAIRCASE.py
n=int(input())
a=list(map(int,input().split()))
a.insert(0,0)
kt=True
f=[0]*1000000
for i in range(1,n+1):
    f[i]=a[1]+i-1
    if f[i]<a[i]:
        kt=False
f[n]=a[n]
if kt==False or f[n]>f[n-1]+1:
    print('impossible')
else:
    for i in range(1,n+1):
        print(min(f[i],f[n]),end=' ')

File: LQD/MAKESTAIRCASE.pas
const no='impossible';
      maxn=trunc(2e5);
var a,f:array[1..maxn] of longint;
    i,n:longint;
    kt:boolean;
function min(a,b:longint):longint;
begin
  if a<b then exit(a)
   else exit(b);
end;
begin
   readln(n);
   kt:=true;
  for i:=1 to n do 
    begin
       read(a[i]);
       f[i]:=a[1]+i-1;
      if f[i]<a[i] then kt:=false;
    end;
   f[n]:=a[n];
  if (f[n]>f[n-1]+1) or (kt=false) then write(no)
   else
     for i:=1 to n do write(min(f[i],f[n]),' ');
end.